flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/join/LookupJoinTest.scala

```diff
@@ -525,7 +525,21 @@ class LookupJoinTest(legacyTableSource: Boolean) extends TableTestBase with Seri
     verifyTranslationSuccess(sql)
   }
 
-  // ==========================================================================================
+  @Test
+  def testJoinTemporalTableWithCTE(): Unit = {
+    val sql =
+      """
+        |WITH MyLookupTable AS (SELECT * FROM MyTable),
+        |OtherLookupTable AS (SELECT * FROM LookupTable)
+        |SELECT MyLookupTable.b FROM MyLookupTable
+        |JOIN OtherLookupTable FOR SYSTEM_TIME AS OF MyLookupTable.proctime AS D
+        |ON MyLookupTable.a = D.id AND D.age = 10
+      """.stripMargin
+
+    verifyTranslationSuccess(sql)
```

Comment:
I would recommend `util.verifyExecPlan` so that we can compare the exact plan in the future.

Note that to use this checking method, you'll have to first remove the corresponding LookupJoinTest.xml file of this test then run the whole test class. You'll find that a new LookupJoinTest.xml is generated and then git will show you what is added to the new LookupJoinTest.xml. Make sure that the added plan meets your expectation.

Reply:
@tsreaper I can add an additional test that verifies to exec plan, as it was similar before and after my change (I originally tried using it in my test, but the `NullPointerException` did not reproduce using it.

Reply:
@YuvalItzchakov I can reproduce the `NullPointerException` with `util.verifyExecPlan`. Maybe you forget to revert the changes before testing.

```suggestion
    util.verifyExecPlan(sql)
```